job_executor/model/metadata.py

```diff
@@ -195,14 +193,14 @@ def dict(self, **kwargs) -> dict:
             "dataType": self.data_type,
```

Comment:
I think we can delete the dict method now as the move to model_dump means we dont use this override anymore 🤔 Do we use the dict method on metadata anywhere in this branch?

Reply:
Some confusion from my part on what was in fact "our own" dict - method, and what was pydantic dict method (deprecated) and where to use what (they were a bit entangled..). 

Some of the dict methods defined in our code looked scary to remove- therefore we chose to keep them all for now. Me and @pawbu went through the code and were able to identify the places where we in fact use our own dict method, and reverted the code back to .dict instead of .model_dump.

So I think it should be good now 👍🏻

Reply:
The unit tests check for this quite a bit as we have metadata json files that we expect to be the output etc.
Even though breaking the conversion of the metadata model to json is scary (it is how we save our metadata after all) the dict methods do not do any implicit magic. 
After reading through them it seems they were created mainly to avoid having keys that have None values being rendered as `"key": null` in the json file. The model dump method now has an "exclude_none" option that you are using that solves this 👍🏻 these options also propagate down through the models.

We dont necessarily need to do away with the dict methods in this PR. But I would like to do away with them now that pydantic is able to do the "jsonification" for us. less code for a pretty verbose model means less code to read in the future 😅 

We can do this in a separate PR and put it up as a separate subtask maybe? What do you think?

Reply:
See new commit!